docs/DeveloperGuide.md

```diff
@@ -1,34 +1,357 @@
+
+
+A single instance of common objects are usually created in UserSession, such as Ui and Printer objects. 
+These common objects are injected into other objects that need them via the class constructor instead of static methods.
+The common objects include the Ui and  Printer helper classes which provide an organized way to read and write data. 
+Most of the programmes’ output is made via calls to the same common printer object.
+
+#### 3.2.2. Command Parser (Brandon)
+
+This component would parse the user input to produce useful information which would be used to construct a Command and executed the command. 
+
+![](Images/Image6.PNG)
+
+Image 4: Interaction between components to parse and execute command
+
+![](Images/Image7.PNG)
+
+Image 5: Sequence Diagram from parsing input to command execution
+
+Image 4 and 5 illustrates the interactions between Parser, CommandExecutor and Command when a user inputs an instruction. 
+
+The steps below explain the sequence diagram:
+1. User input an instruction
+2. A new Parser object is created
+3. Parser#parse() is called to extract command type, name, programming language and details from the user-inputted instruction
+4. If the user input is valid, a new CommandExecutor object would be created
+5. CommandExecutor#execute() would be called to create a new Command object according to type of command user inputted
+6. Command#execute() would call other methods from CheatSheet and CheatSheetList to carry out specific instructions.
+
+
+#### 3.2.3. Cheat Sheet Structure (Aldo)
+
+To further understand our cheatsheet management application, it is important to understand the inner workings of the CheatSheet class. 
+CheatLogs comes with preloaded cheat sheets and the application has an in-built ability for the user to add, edit, and delete their own cheat sheets. 
+The preloaded cheat sheet cannot be modified by the user, as the user-made cheatsheet is able to be modified by the user.
+
+
+The CheatSheet class is the class that models the cheat sheet that we used in our program. 
+Each of our cheatsheet object is intended to store one cheatsheet topic for one specific programming language (both preloaded and user-created cheat sheets).
+
+This system is intended to increase user access and organization of the cheat sheets.
+Each of the CheatSheet object has:
+1. 	Name
+2. 	Programming Language
+3. 	Description (Content of the cheatsheet)
+4. 	isFavorite
+5. 	isModifiable
+6. 	Date Created
+7. 	Date Modified
+
+![](Images/Image8.PNG)
+
+Image 6: class diagram of the cheatsheet class
+
+##### Development Goals
+
+We are trying to move our existing cheatsheet file format from txt to xml. 
+XML files provide inbuilt tags to differentiate and classify members of the document. 
+With the use of XML file, it simplifies the parsing process of the cheatsheet files into the cheatsheet class, and vice versa.
+
+![](Images/Image9.PNG)
+
+Image 7: XML file showing the content of a cheatsheet file
```

Comment:
There is a missing reference to this image. Maybe you could add a reference in the paragraph above?